### PR TITLE
Add Block Editor Handbook page template

### PR DESCRIPTION
**Pull request by ryelle**
This is a first-pass at adding in the `single.html` from Documentation, for use on the handbook pages. Right now this only applies to the Block Editor Handbook, I don't have any page content in the other handbooks locally to test. This is a start for #161, but does not fix it because we still need the left column handbook nav. 

This also serves as a demo for https://github.com/WordPress/wporg-mu-plugins/pull/321.

**Screenshots**

| Desktop | Mobile |
|-------|------|
| fullpage | mobile |

**To test**

1. View a block editor handbook page, like `http://localhost:8888/block-editor/getting-started/create-block/block-code/`

To remove the old Table of Contents I added the same filter that [I added to the support-helphub plugin](https://meta.trac.wordpress.org/changeset/12327/) which uses this same code (not literally, unfortunately). Replace `load_filters` in `source/wp-content/plugins/handbook/inc/table-of-contents.php` with this:

```php
public function load_filters() {
	$page_supports_toc = is_singular( $this->post_types ) && ! is_embed() && ! is_front_page();
	/**
	 * Filter whether the table of contents should be injected into the page.
	 *
	 * @param bool $page_supports_toc True if the current page supports a table of contents.
	 */
	$should_add_toc = apply_filters( 'wporg_handbook_toc_should_add_toc', $page_supports_toc );
	if ( $should_add_toc ) {
		add_filter( 'the_content', array( $this, 'add_toc' ) );
	}
}
```

### CODE REVIEW

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			<!-- wp:post-title {"level":1,"style":{"spacing":{"margin":{"top":"var:preset|spacing|edge-space","bottom":"var:preset|spacing|40"}}}} /-->
+
+			<!-- wp:group {"tagName":"aside","className":"sidebar-container"} -->
+			<aside class="wp-block-group sidebar-container">
+				<!-- wp:search {"label":"Search","showLabel":false,"widthUnit":"px","buttonText":"Search","buttonPosition":"button-inside","buttonUseIcon":true,"className":"is-style-secondary-search-control","borderColor":"light-grey-1","placeholder":"Search the documentation"} /-->
+
+				<!-- wp:wporg/table-of-contents {"style":{"elements":{"link":{"color":{"text":"var:preset|color|blueberry-1"}}},"spacing":{"padding":{"top":"var:preset|spacing|20","right":"var:preset|spacing|20","bottom":"var:preset|spacing|20","left":"var:preset|spacing|20"}}},"backgroundColor":"blueberry-4","textColor":"blueberry-1"} /-->
+
+				<!-- wp:paragraph {"fontSize":"small","className":"is-link-to-top"} -->
+				<p class="has-small-font-size is-link-to-top"><a href="#wp--skip-link--target">↑ Back to top</a></p>
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: We'll want this to be translated.

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			<!-- /wp:group -->
+
+			<!-- wp:group {"style":{"spacing":{"blockGap":"0"}},"layout":{"type":"constrained"}} -->
+			<div class="wp-block-group">
+				<!-- wp:paragraph {"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}}} -->
+				<p style="font-style:normal;font-weight:700">Changelog</p>
+				<!-- /wp:paragraph -->
+		
+				<!-- wp:paragraph {"className":"external-link"} -->
+				<p class="external-link"><a href="#">See list of changes</a></p>
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: Are we going to follow up with these links?

File: source/wp-content/themes/wporg-developer-2023/src/style/style.scss [hunk 3]
```diff
@@ -14,3 +19,72 @@
...
+		}
+
+		&.is-fixed-sidebar .is-link-to-top,
+		&.is-bottom-sidebar .is-link-to-top {
+			display: block;
+		}
+	}
+}
+
+.wp-block-search.wp-block-search__button-inside .wp-block-search__inside-wrapper {
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-developer-2023/src/style/style.scss:57]: This is also being added in #164 
https://github.com/WordPress/wporg-developer/pull/164/files#diff-5a967de023463491bebbfaca0909be4be783855fcace73f96a9a0e9c8060224aR18

**Comment by StevenDufresne** [comment]
A couple of minor comments. Works well 👍

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	<!-- wp:group {"layout":{"type":"constrained","justifyContent":"left"},"align":"wide"} -->
+	<div class="wp-block-group alignwide">
+
+		<!-- wp:group {"tagName":"article"} -->
+		<article class="wp-block-group">
+			<!-- wp:post-title {"level":1,"style":{"spacing":{"margin":{"top":"var:preset|spacing|edge-space","bottom":"var:preset|spacing|40"}}}} /-->
+
+			<!-- wp:group {"tagName":"aside","className":"sidebar-container"} -->
+			<aside class="wp-block-group sidebar-container">
+				<!-- wp:search {"label":"Search","showLabel":false,"widthUnit":"px","buttonText":"Search","buttonPosition":"button-inside","buttonUseIcon":true,"className":"is-style-secondary-search-control","borderColor":"light-grey-1","placeholder":"Search the documentation"} /-->
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: This is dropping into the code search results. We'll need to update that at some point.

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html (hunk 2 above)
Comment (ryelle) [reply source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: Yeah, there's still plenty to follow up with — this section also needs mobile styling, for example. I honestly meant to draft this PR but when I didn't I decided to let it be a first-pass & iterate in trunk 😅 

If you'd rather I keep iterating in the branch though, I can swap this into to draft mode.

Comment (StevenDufresne) [reply source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: Nope, merge it in when you think it's ready.

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html (hunk 4 above)
Comment (ryelle) [reply source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: This is all done in a `searchform.php` template in the old theme… we can probably do something like [the global header search](https://github.com/WordPress/wporg-mu-plugins/blob/16ee1d69b809ac6d4e1b6963e0192f2dd4802af9/mu-plugins/blocks/global-header-footer/blocks.php#L999-L1016).

At least it looks like the search is mostly works as long as we set the form URL correctly:

Screen Shot 2023-01-24 at 16 13 26
http://localhost:8888/block-editor/?s=test

File: source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html (hunk 1 above)
Comment (ryelle) [reply source/wp-content/themes/wporg-developer-2023/templates/single-blocks-handbook.html]: I've moved everything into a pattern and wrapped all these in translation functions. This content isn't synced from anywhere, so it's all been manually wrapped in i18n functions.

File: source/wp-content/themes/wporg-developer-2023/patterns/single-handbook.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+			<!-- /wp:group -->
+
+			<!-- wp:group {"style":{"spacing":{"blockGap":"0"}},"layout":{"type":"constrained"}} -->
+			<div class="wp-block-group">
+				<!-- wp:paragraph {"style":{"typography":{"fontStyle":"normal","fontWeight":"700"}}} -->
+				<p style="font-style:normal;font-weight:700"><?php esc_html_e( 'Edit article', 'wporg' ); ?></p>
+				<!-- /wp:paragraph -->
+
+				<!-- wp:paragraph {"className":"external-link"} -->
+				<p class="external-link"><a href="#"><?php esc_html_e( 'Improve it on GitHub', 'wporg' ); ?></a></p>
```
Comment (StevenDufresne) [inline source/wp-content/themes/wporg-developer-2023/patterns/single-handbook.php:65]: How are these generated?

**Review by StevenDufresne — APPROVED** [review]
Good start!

File: source/wp-content/themes/wporg-developer-2023/patterns/single-handbook.php (hunk 5 above)
Comment (ryelle) [reply source/wp-content/themes/wporg-developer-2023/patterns/single-handbook.php:65]: It looks like this _would_ be generated just by the `get_edit_post_link` function, but due to how patterns are rendered the current post isn't available yet, so we can't just use that function. We'll probably need to create some shortcodes for these links.

Comment (ryelle) [reply source/wp-content/themes/wporg-developer-2023/patterns/single-handbook.php:65]: And I don't think code to generate those other links exists at the moment, but I didn't look very deeply.

**Pull request merged**